File: 191-NumberOf1Bits/191-NumberOf1Bits.php
// Last updated: 5/10/2026, 11:52:44 PM
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function hammingWeight($n) {
        $count = 0;
        while ($n > 0) {
            if ($n % 2 == 1) $count++;
            $n = intdiv($n, 2);
        }
        return $count;
    }
}

File: 191-NumberOf1Bits/191-NumberOf1Bits.cpp
// Last updated: 5/10/2026, 11:52:50 PM
class Solution {
public:
    int hammingWeight(int n) {
        int count = 1;
        while(n > 1){
            if(n%2 == 1)count++;
            n /= 2;
        }
        return count;
    }
};